src/gcodePlanner.cpp

```diff
@@ -409,15 +409,23 @@ void GCodePlanner::addPolygonsByOptimizer(Polygons& polygons, GCodePathConfig* c
     {
         return;
     }
-    PathOrderOptimizer orderOptimizer(lastPosition, z_seam_pos, z_seam_type);
-    for (unsigned int poly_idx = 0; poly_idx < polygons.size(); poly_idx++)
+    if (spiralize)
     {
-        orderOptimizer.addPolygon(polygons[poly_idx]);
+        // only interested in the outer perimeter
+        addPolygon(polygons[0], 0, config, wall_overlap_computation, wall_0_wipe_dist, spiralize);
```

Comment:
Spiralize is a heavily limited option. You shouldnt use it on models like the one you showed. The models should be solid.

I do like your thought, though. I was thinking of making a frontend setting out of that.